### PR TITLE
Add error in provisioning screens

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

We never ported over the use of this screen to the new login saga stuff.

This PR:

* Shares the error in provisioning screen across mobile/desktop
* Navigates to that error screen when we get an error in the finishedSaga

### CODE REVIEW

File: shared/actions/login/saga.js [hunk 1]
```diff
@@ -403,14 +403,25 @@ function addDeviceRpc (channelConfig) {
...
 }
 
+function * handleProvisioningError (error) {
+  yield put(navigateAppend([{
+    props: {
+      error,
+    },
+    selected: 'error',
+  }], [loginTab, 'login']))
+  yield take(Constants.onBack)
```
Comment (MarcoPolo) [inline shared/actions/login/saga.js]: The right way here is to have the error screen dispatch a different action that causes a saga to run `navBasedOnLoginState`. This was just easier

**Comment by cjb** [comment]
> Shares the error in provisioning screen across mobile/desktop

Hm, if you have a GPG key that isn't synced with server, and no devices, then you *can* log in from the desktop if you set up GPG properly, but you *can't* log in from mobile because there's no GPG, so we need to tell you to go find a desktop app install instead.

Maybe we can fix the text to work that way, if it doesn't now?

**Comment by MarcoPolo** [comment]
We say:

` - Install Keybase on a machine that has your PGP private key in it` if we can't find a matching gpg

and 

` - Install Keybase on a different machine that has your PGP key` if we can't find a matching gpg key and no other active device.

We also suggest  to use the command line to log in too.

**Comment by keybase-ci-visdiff** [comment]
The commits 8e9416578be4...fdeda9ee1f3a introduce visual changes on linux. <details><summary>:mag_right: 4 changed</summary>

 * Changed: **Error_View-Key_No_Secret** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494969792414-8e9416578be4-fdeda9ee1f3a-linux/Error_View-Key_No_Secret-8e9416578be4.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494969792414-8e9416578be4-fdeda9ee1f3a-linux/Error_View-Key_No_Secret-fdeda9ee1f3a.png)  
   Error_View-Key_No_Secret comparison
 * Changed: **Error_View-Key_No_Matching_GPG** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494969792414-8e9416578be4-fdeda9ee1f3a-linux/Error_View-Key_No_Matching_GPG-8e9416578be4.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494969792414-8e9416578be4-fdeda9ee1f3a-linux/Error_View-Key_No_Matching_GPG-fdeda9ee1f3a.png)  
   Error_View-Key_No_Matching_GPG comparison
 * Changed: **Error_View-Device_No_Provision** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494969792414-8e9416578be4-fdeda9ee1f3a-linux/Error_View-Device_No_Provision-8e9416578be4.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494969792414-8e9416578be4-fdeda9ee1f3a-linux/Error_View-Device_No_Provision-fdeda9ee1f3a.png)  
   Error_View-Device_No_Provision comparison
 * Changed: **Error_View-Bad_Login_Password** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494969792414-8e9416578be4-fdeda9ee1f3a-linux/Error_View-Bad_Login_Password-8e9416578be4.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494969792414-8e9416578be4-fdeda9ee1f3a-linux/Error_View-Bad_Login_Password-fdeda9ee1f3a.png)  
   Error_View-Bad_Login_Password comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 8e9416578be4...b0a1a931d6ba introduce visual changes on linux. <details><summary>:mag_right: 4 changed</summary>

 * Changed: **Error_View-Key_No_Secret** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494970162163-8e9416578be4-b0a1a931d6ba-linux/Error_View-Key_No_Secret-8e9416578be4.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494970162163-8e9416578be4-b0a1a931d6ba-linux/Error_View-Key_No_Secret-b0a1a931d6ba.png)  
   Error_View-Key_No_Secret comparison
 * Changed: **Error_View-Key_No_Matching_GPG** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494970162163-8e9416578be4-b0a1a931d6ba-linux/Error_View-Key_No_Matching_GPG-8e9416578be4.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494970162163-8e9416578be4-b0a1a931d6ba-linux/Error_View-Key_No_Matching_GPG-b0a1a931d6ba.png)  
   Error_View-Key_No_Matching_GPG comparison
 * Changed: **Error_View-Device_No_Provision** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494970162163-8e9416578be4-b0a1a931d6ba-linux/Error_View-Device_No_Provision-8e9416578be4.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494970162163-8e9416578be4-b0a1a931d6ba-linux/Error_View-Device_No_Provision-b0a1a931d6ba.png)  
   Error_View-Device_No_Provision comparison
 * Changed: **Error_View-Bad_Login_Password** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494970162163-8e9416578be4-b0a1a931d6ba-linux/Error_View-Bad_Login_Password-8e9416578be4.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494970162163-8e9416578be4-b0a1a931d6ba-linux/Error_View-Bad_Login_Password-b0a1a931d6ba.png)  
   Error_View-Bad_Login_Password comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 8e9416578be4...50398a06e540 introduce visual changes on linux. <details><summary>:mag_right: 4 changed</summary>

 * Changed: **Error_View-Key_No_Secret** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1495043490017-8e9416578be4-50398a06e540-linux/Error_View-Key_No_Secret-8e9416578be4.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1495043490017-8e9416578be4-50398a06e540-linux/Error_View-Key_No_Secret-50398a06e540.png)  
   Error_View-Key_No_Secret comparison
 * Changed: **Error_View-Key_No_Matching_GPG** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1495043490017-8e9416578be4-50398a06e540-linux/Error_View-Key_No_Matching_GPG-8e9416578be4.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1495043490017-8e9416578be4-50398a06e540-linux/Error_View-Key_No_Matching_GPG-50398a06e540.png)  
   Error_View-Key_No_Matching_GPG comparison
 * Changed: **Error_View-Device_No_Provision** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1495043490017-8e9416578be4-50398a06e540-linux/Error_View-Device_No_Provision-8e9416578be4.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1495043490017-8e9416578be4-50398a06e540-linux/Error_View-Device_No_Provision-50398a06e540.png)  
   Error_View-Device_No_Provision comparison
 * Changed: **Error_View-Bad_Login_Password** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1495043490017-8e9416578be4-50398a06e540-linux/Error_View-Bad_Login_Password-8e9416578be4.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1495043490017-8e9416578be4-50398a06e540-linux/Error_View-Bad_Login_Password-50398a06e540.png)  
   Error_View-Bad_Login_Password comparison
</summary>

**Pull request merged**